### PR TITLE
KAN-250: Refactor Student Model: Migration and Redefinition

### PR DESCRIPTION
**Pull request by JPraskala**
This PR introduces significant changes related to the `Student` model in the application. The following changes have been made:

- **Migration Files Created:**
  - **0065_delete_student.py:**
    - This migration file deletes the existing `Student` model from the database.
  - **0066_student.py:**
    - This migration file re-defines the `Student` model. The model now has the following properties:
      - A UUID primary key field.
      - A one-to-one relationship with the `User` model.

- **Model Definition Update:**
  - In `models.py`:
    - The `Student` model has been redefined to include a UUID primary key (`id`) using `uuid.uuid4()`, ensuring each student has a unique identifier.
    - The one-to-one field `user` for user association is retained, providing a link to the authenticated user.

This update aims to improve data integrity and align the `Student` model with updated best practices by utilizing UUIDs for unique identification.

### CODE REVIEW

**Comment by JPraskala** [comment]
I have made the migrations, but I have not migrated them yet. I am waiting for approval before I do so.

**Review by amhajja — APPROVED** [review]
Looks great!

**Pull request merged**
